src/test/java/com/google/sps/data/RecommendationEngineTest.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import java.util.*;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class RecommendationEngineTest {
+  private Drink DRINKA = new Drink("latte", "123", 4.8, 12, null);
+  private Drink DRINKB = new Drink("cold brew", "123", 4.8, 12, null);
+  private Drink DRINKC = new Drink("latte", "456", 4.2, 12, null);
+  private Drink DRINKD = new Drink("cold brew", "456", 4.2, 12, null);
+  private Drink DRINKE = new Drink("latte", "789", 1.5, 12, null);
+  private Drink DRINKF = new Drink("cold brew", "789", 1.5, 12, null);
+  private List<Drink> drinkList = new ArrayList<>();
+  private static final double DELTA = .0000001;
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testGetBestShop() {
+    drinkList.add(DRINKA);
+    drinkList.add(DRINKB);
+    drinkList.add(DRINKC);
+    drinkList.add(DRINKD);
+    drinkList.add(DRINKE);
+    drinkList.add(DRINKF);
+    List<String> idList = new ArrayList<>();
+    for (Drink drink : drinkList) {
+      idList.add(drink.getStore());
+    }
+    Assert.assertEquals("123", RecommendationEngine.getBestShop(idList, "latte", drinkList));
```

Comment:
In general, prefer not to have magic numbers in your code. Is "123" a store Id? If yes, suggest specifying storeIdA = "123", and using the human-readable variable name in the test. Makes it easier to understand the test's intent.

Reply:
i just added a variable to contain 123 and have inserted that instead